src/function/scalar/string/left_right.cpp

```diff
@@ -0,0 +1,44 @@
+#include "duckdb/function/scalar/string_functions.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "util.h"
+
+#include <string.h>
+#include <ctype.h>
+#include <algorithm>	// std::max and std::min
+
+#include <iostream>
+
+using namespace std;
+
+namespace duckdb {
+
+static string_t left_scalar_function(Vector &result, const string_t str, int64_t pos) {
+	idx_t current_len = 0;
+	unique_ptr<char[]> output;
```

Comment:
The idea behind having this output be a parameter of the function is to define it outside of the loop so the buffer can be reused, which can reduce the number of allocations made significantly.

Can you move the `output` buffer and the `current_len` to the base of `left_function` below, before the BinaryExecutor::Execute?

Reply:
Thanks for reminding, comment addressed, PTAL.